### PR TITLE
chore: move Kint to `require-dev`

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Kint is not needed on production environment.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Helpers/kint_helper.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            Kint::dump(...$vars);
+
+            exit;
+            // @codeCoverageIgnoreEnd
+        }
+    } else {
+        // In case that Kint is not loaded.
+        function dd(...$vars)
+        {
+            return 0;
```
Comment (MGatner) [inline system/Helpers/kint_helper.php:36]: This is potentially dangerous. If someone has been using `dd()` in production then they are likely relying on it terminating. I would propose a closer fallback:
```suggestion
            var_dump(...$vars);
            exit;
```

File: system/Helpers/kint_helper.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+}
+
+if (! function_exists('d') && ! class_exists(Kint::class)) {
+    // In case that Kint is not loaded.
+    function d(...$vars)
+    {
+        return 0;
+    }
```
Comment (MGatner) [inline system/Helpers/kint_helper.php:46]: Why not approximate?
```suggestion
            var_dump(...$vars);
```

File: system/Helpers/kint_helper.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            Kint::$aliases[] = 'trace';
+            Kint::trace();
+        }
+    } else {
+        // In case that Kint is not loaded.
+        function trace()
+        {
+            return 0;
+        }
+    }
```
Comment (MGatner) [inline system/Helpers/kint_helper.php:65]:
```suggestion
            var_dump(debug_backtrace());
```

**Review by MGatner — CHANGES_REQUESTED** [review]
The `dd` comment is important; others are up for discussion.

File: system/Helpers/kint_helper.php (hunk 1 above)
Comment (kenjis) [reply system/Helpers/kint_helper.php:36]: Kint is disabled in production environment:
https://github.com/codeigniter4/CodeIgniter4/blob/2788a492ea9b2ae8c42873ed01ecf435df8d269c/system/CodeIgniter.php#L188-L190

So `dd()` returns `0`:
https://github.com/codeigniter4/CodeIgniter4/blob/2788a492ea9b2ae8c42873ed01ecf435df8d269c/system/ThirdParty/Kint/Kint.php#L558-L560

Comment (MGatner) [reply system/Helpers/kint_helper.php:36]: Ah that makes sense! Thanks

**Review by MGatner — APPROVED** [review]

**Pull request merged**